xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/skin/EnvironmentSkin.java

```diff
@@ -69,91 +71,43 @@ public EnvironmentSkin(String id, InternalSkinManager skinManager, InternalSkinC
-    {
-        if (this.properties == null) {
-            URL url = this.environment.getResource(getSkinFolder() + "skin.properties");
-            if (url != null) {
-                try {
-                    this.properties = new Configurations().properties(url);
-                } catch (ConfigurationException e) {
-                    LOGGER.error("Failed to load skin [{}] properties file ([])", this.id, url,
-                        ExceptionUtils.getRootCauseMessage(e));
-
-                    this.properties = new BaseConfiguration();
-                }
-            } else {
-                LOGGER.debug("No properties found for skin [{}]", this.id);
-
-                this.properties = new BaseConfiguration();
-            }
-        }
-
-        return this.properties;
-    }
-
-    public String getSkinFolder()
+    protected URL getResourceURL(String propertiesPath)
     {
-        return "/skins/" + this.id + '/';
+        return this.environment.getResource(propertiesPath);
     }
 
     @Override
-    public Resource<?> getLocalResource(String resourceName)
-    {
-        String resourcePath = getSkinResourcePath(resourceName);
-
-        if (this.environment.getResource(resourcePath) != null) {
-            return createResource(resourcePath, resourceName);
-        }
-
-        return null;
-    }
-
-    protected AbstractEnvironmentResource createResource(String resourcePath, String resourceName)
+    protected SkinEnvironmentResource createResource(String resourcePath, String resourceName)
     {
         return new SkinEnvironmentResource(resourcePath, resourceName, this, this.environment, this.xcontextProvider,
             this.urlConfiguration);
     }
 
-    private String getSkinResourcePath(String resource)
+    @Override
+    protected String getSkinFolder()
     {
-        String skinFolder = getSkinFolder();
-        String resourcePath = skinFolder + resource;
-
-        // Prevent inclusion of templates from other directories
-        Path normalizedResource = Paths.get(resourcePath).normalize();
-        if (!normalizedResource.startsWith(skinFolder)) {
-            LOGGER.warn("Direct access to skin file [{}] refused. Possible break-in attempt!", normalizedResource);
-
-            return null;
-        }
-
-        return resourcePath;
+        return '/' + super.getSkinFolder();
     }
 
-    @Override
-    public String getOutputSyntaxString()
+    /**
+     * Check if the skin exists by checking if a directory exists and contains a {@code skin.properties} file.
+     *
+     * @return {@code true} if the skin exists, {@code false} otherwise
+     */
+    public boolean exists()
     {
-        return getProperties().getString("outputSyntax");
+        String propertiesPath = getPropertiesPath();
+        URL resource = getResourceURL(propertiesPath);
+        if (resource == null) {
+            return false;
+        }
+        try {
+            return Paths.get(resource.toURI()).toFile().exists();
```

Comment:
Are you sure you need that ? AFAIK getResourceURL will return null if the file does not exist already.